app/src/main/java/org/mozilla/tv/firefox/pocket/PocketRepo.kt

```diff
@@ -23,4 +35,72 @@ open class PocketRepo(private val pocketEndpoint: PocketEndpoint) {
     private val mutableState = MutableLiveData<PocketRepoState>()
     open val state: LiveData<PocketRepoState> = mutableState
 
+    @Suppress("ANNOTATION_TARGETS_NON_EXISTENT_ACCESSOR") // Private properties generate fields so method annotations can't apply.
+    @get:UiThread
+    @set:UiThread
+    private var backgroundUpdates: Job? = null
+    @Volatile private var lastUpdateMillis = -1L
+    private val nextUpdateMillis get() = lastUpdateMillis + CACHE_UPDATE_FREQUENCY_MILLIS
+
+    init {
+        // mutableState.value should always be initialized at the top of init, because we treat
+        // it as non-null
+        @Suppress("SENSELESS_COMPARISON")
+        // Values of BuildConfig can change but the compiler doesn't know that
+        mutableState.value = when {
+            BuildConfig.POCKET_KEY == null -> PocketRepoState.NoKey
+            else -> PocketRepoState.Loading
+        }
+
+        update()
```

Comment:
I don't like that this has side effects in the constructor: this generally mocks mocking difficult. I'd rather have a state factory method has the initialization method that constructs and calls `update()`.